File: cpp/0035搜索插入位置.cpp
#include<iostream>
#include<vector>

using namespace std;

int SearchInsert(vector<int> nums,int target)
{
    if(nums.size()==0)
        return 0;

    int left = 0;
    int right = nums.size() - 1;
    int mid;
    while (left <= right)
    {
        mid = (left + right) >>1;
        if(nums[mid]==target)
            return mid;
        if(nums[mid]>target)
            right = mid - 1;
        else
            left = mid + 1;
    }

    return target <= nums[mid] ? mid : mid + 1;
}

int main()
{
    vector<int> nums = {1, 3, 5, 6};
    
    int target = 7;
    int a=SearchInsert(nums, target);
    cout << a;
}

File: python/main/0035搜索插入位置.py
nums=[1, 3, 5, 6]
target=7

# 二分查找
def SearchInsert(nums,target):
    if(not nums):
        return 0
    left=0
    right=len(nums)-1

    while(left<=right):
        mid=(left+right)//2

        if(nums[mid]==target):
            return mid
        if(nums[mid]>target):
            right=mid-1
        else:
            left=mid+1
    
    return mid+1 if nums[mid]<target else mid


print(SearchInsert(nums,target))

    